packages/core/client/src/schema-component/antd/select/utils.ts

```diff
@@ -45,9 +45,9 @@ function flatData(data: any[], fieldNames: FieldNames): any[] {
 
 export function getCurrentOptions(values: string | string[], dataSource: any[], fieldNames: FieldNames): Option[] {
   const result = flatData(dataSource, fieldNames);
-  const arrValues = castArray(values)
-    .filter((item) => item != null)
-    .map((val) => (isPlainObject(val) ? val[fieldNames.value] : val)) as string[];
+  const arrValues = castArray(values).map((val) =>
+    isPlainObject(val) && val !== null ? val[fieldNames.value] : val,
```

Comment:
```suggestion
    isPlainObject(val) ? val[fieldNames.value] : val,
```
如果确定是一个 object，那就一定不会等于 null 了。所以这个 `val !== null` 可以去掉。